packages/gatsby-plugin-sharp/src/gatsby-node.js

```diff
@@ -2,10 +2,26 @@ const {
   setBoundActionCreators,
   // queue: jobQueue,
   // reportError,
+  _unstable_createJob,
 } = require(`./index`)
+const { pathExists } = require(`fs-extra`)
+const { slash } = require(`gatsby-core-utils`)
+const { trackFeatureIsUsed } = require(`gatsby-telemetry`)
 const { getProgressBar, createOrGetProgressBar } = require(`./utils`)
 
 const { setPluginOptions } = require(`./plugin-options`)
+const path = require(`path`)
+
+exports.onPreInit = ({ reporter }) => {
+  if (process.env.GATSBY_EXPERIMENTAL_LAZY_IMAGES) {
+    if (!process.env.GATSBY_EXPERIMENTAL_FAST_DEV) {
+      reporter.info(
+        `[gatsby-plugin-sharp] The lazy image processing experiment is enabled`
+      )
+    }
+    trackFeatureIsUsed(`LazyImageProcessing`)
+  }
+}
```

Comment:
We need to add check for command here, because we do print the message (and do telemetry tracking) in builds now (even tho we actually don't have lazy processing in builds we guard against that)

https://github.com/gatsbyjs/gatsby/blob/c993d52bff1b2255ba8a2c0d04726bfdb71b070a/packages/gatsby/src/services/initialize.ts#L216-L226

Is "prior art" - the CI part that I included in selected lines is not something that I can decide if it should be included (disable this on CI or not here), but just something to consider

Reply:
So this is just a guard against a user error (when the user sets this env variable for any environment other than develop)?

Reply:
Yup, it has 2 uses:
 1) it disables the notice in terminal in non develop commands
 2) it shields the cases where implementation wise we (developers) miss checking if command is `develop` and only check if env var is set ( so this is not required per se, but it feels much safer for something that is very easy to miss when implementing or reviewing )

Reply:
so while setting env var on non-develop might be "user error" - we should not let this break things ;)

Reply:
we can automatically support this sort of nice behavior as well in flags in the future — it asks you to say which env a flag is enabled in and then could disable it otherwise.